internal/cloud/atlas/groups_test.go

```diff
@@ -49,6 +51,28 @@ func TestAtlasGroups(t *testing.T) {
 	})
 }
 
+func TestFetchGroups(t *testing.T) {
+	u.SkipUnlessAtlasServerRunning(t)
+
+	t.Run("should fetch all groups", func(t *testing.T) {
+		for _, tc := range []struct {
+			numGroups int
+		}{
+			{numGroups: 1},
+			{numGroups: 3},
+			{numGroups: 5},
+		} {
+			t.Run(fmt.Sprintf("when num items allowed per page is %d", tc.numGroups), func(t *testing.T) {
+				client := newAuthClient(t)
+				options := api.RequestOptions{Query: map[string]string{"itemsPerPage": fmt.Sprint(tc.numGroups)}}
+				groups, err := client.FetchGroups(u.AtlasServerURL()+"/api/public/v1.0/groups", options)
```

Comment:
I think you could theoretically get around this "exporting a function for testing purposes" by mocking `cloud.Client` here and returning a chain of responses (per call) that force you to iteratively fetch the expected number of groups

it doesn't have to be "live data" to prove that your recursion is working here (which is ultimately what this test wants to do)

Reply:
ignore me... doing this sort of approach would either retweak the entire `Atlas` client interface or doing the same "export something for testing purposes" here.  not worth it, this (indirectly) proves our recursion is working as expected as well